src/app/shared/utils/utils.ts

```diff
@@ -49,4 +51,40 @@ export class Utils {
     .replace(/%5C/g, '_')  // Replace encoded backslash with underscore
     .replace(/%20/g, '_'); // Replace encoded space with underscore
   }
+
+  /**
+   * Returns an estimated connection tier based on downlink.
+   * If downlink can't be found, it falls back to small image load time.
+   * 
+   * @returns {Promise<ConnectionTier>}
+   */
+  public async getConnectionTier(): Promise<ConnectionTier> {
+    // Try to retrieve the connection speed directly
+    const conn = (navigator as any).connection || (navigator as any).mozConnection || (navigator as any).webkitConnection;
+
+    if (conn?.downlink) {
+      const speed = conn.downlink;
+      if (speed < 1) return 'slow';
+      else if (speed < 10) return 'medium';
+      else if (speed < 100) return 'fast';
+      else return 'turbo';
+    }
+
+    // Image load time fallback
+    return new Promise<ConnectionTier>((resolve) => {
+      const img = new Image();
```

Comment:
So each time the map moves or zooms, and image is being created and loaded in memory and a setTimeout is fired. I feel this could be a performance concern

Reply:
No, that's incorrect. Both events are fired with .once now, meaning they're only used once.